I'm not sure why typing in Chinese with the input method causes Zed to crash.

### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I don't know where the error occurred and caused Zed to crash. 

### Environment

Windows 10 22H2 19045.4957
Ram: 32GB
CPU: Intel(R) Core(TM) i9-14900HX   2.20 GHz

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->
2024-10-11T09:55:03.1005807+08:00 [INFO] ========== starting zed ==========
2024-10-11T09:55:03.2392089+08:00 [INFO] Use Microsoft YaHei UI as UI font.
2024-10-11T09:55:03.2406705+08:00 [INFO] Opening main db
2024-10-11T09:55:03.2435792+08:00 [INFO] Opening main db
2024-10-11T09:55:03.2469591+08:00 [INFO] Using git binary path: None
2024-10-11T09:55:03.2477182+08:00 [ERROR] SHELL environment variable is not assigned so we can't source login environment variables

Caused by:
    environment variable not found
2024-10-11T09:55:03.2721088+08:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-10-11T09:55:03.2723951+08:00 [INFO] initializing context server client
2024-10-11T09:55:03.2726746+08:00 [INFO] Registering built-in prompt template: content_prompt
2024-10-11T09:55:03.2732595+08:00 [INFO] Registering built-in prompt template: edit_workflow
2024-10-11T09:55:03.2734674+08:00 [INFO] Registering built-in prompt template: project_slash_command
2024-10-11T09:55:03.2735687+08:00 [INFO] Registering built-in prompt template: step_resolution
2024-10-11T09:55:03.2739507+08:00 [INFO] Registering built-in prompt template: terminal_assistant_prompt
2024-10-11T09:55:03.2743484+08:00 [INFO] No prompt template overrides directory found at C:\Users\zero\AppData\Local\Zed\prompt_overrides. Using built-in prompts.
2024-10-11T09:55:03.2750298+08:00 [INFO] extensions updated. loading 1, reloading 0, unloading 0
2024-10-11T09:55:03.2824054+08:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-10-11T09:55:03.2836934+08:00 [INFO] Opening main db
2024-10-11T09:55:03.287046+08:00 [INFO] set status on client 0: Authenticating
2024-10-11T09:55:03.2871862+08:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-10-11T09:55:03.3234933+08:00 [INFO] Enabling Vulkan Portability
2024-10-11T09:55:03.3235402+08:00 [INFO] Enabling color space support
2024-10-11T09:55:03.3452535+08:00 [INFO] Adapter: "NVIDIA GeForce RTX 4060 Laptop GPU"
2024-10-11T09:55:03.34562+08:00 [INFO] Ray tracing is supported
2024-10-11T09:55:03.6028052+08:00 [WARN] Requested size 1x1 is outside of surface capabilities
2024-10-11T09:55:03.6028509+08:00 [ERROR] No composite alpha flag for transparency: OPAQUE
2024-10-11T09:55:03.6029791+08:00 [INFO] Using surface present mode MAILBOX
2024-10-11T09:55:03.6065892+08:00 [WARN] Unable to forbid exclusive full screen
2024-10-11T09:55:03.6127626+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-10-11T09:55:03.6128006+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-10-11T09:55:03.6128332+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-10-11T09:55:03.6142002+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6144355+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6150417+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6152459+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6156257+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6157778+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-10-11T09:55:03.6158085+08:00 [INFO] Skip function Some("to_device_position")
2024-10-11T09:55:03.6158385+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-10-11T09:55:03.6158684+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6159315+08:00 [INFO] Skip function Some("blend_color")
2024-10-11T09:55:03.6163181+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6166007+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6167472+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6171675+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6176069+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6178996+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6180897+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-10-11T09:55:03.6181215+08:00 [INFO] Skip function Some("to_device_position")
2024-10-11T09:55:03.618154+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-10-11T09:55:03.6181846+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6182559+08:00 [INFO] Skip function Some("blend_color")
2024-10-11T09:55:03.6193385+08:00 [ERROR] No composite alpha flag for transparency: OPAQUE
2024-10-11T09:55:03.6194345+08:00 [INFO] Using surface present mode MAILBOX
2024-10-11T09:55:03.6224107+08:00 [WARN] Unable to forbid exclusive full screen
2024-10-11T09:55:03.6258836+08:00 [INFO] Using surface present mode MAILBOX
2024-10-11T09:55:03.6275632+08:00 [WARN] Unable to forbid exclusive full screen
2024-10-11T09:55:03.6295086+08:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-10-11T09:55:03.630659+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6308617+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.631211+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6313801+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6316939+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6318611+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-10-11T09:55:03.6318972+08:00 [INFO] Skip function Some("to_device_position")
2024-10-11T09:55:03.6319326+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-10-11T09:55:03.6319681+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6320474+08:00 [INFO] Skip function Some("blend_color")
2024-10-11T09:55:03.6324001+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6326429+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6328038+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6332554+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6336498+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6339742+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6341578+08:00 [INFO] Skip function Some("to_device_position_impl")
2024-10-11T09:55:03.6341922+08:00 [INFO] Skip function Some("to_device_position")
2024-10-11T09:55:03.6342257+08:00 [INFO] Skip function Some("to_device_position_transformed")
2024-10-11T09:55:03.6342592+08:00 [INFO] Skip function Some("to_tile_position")
2024-10-11T09:55:03.6343405+08:00 [INFO] Skip function Some("blend_color")
2024-10-11T09:55:03.6350952+08:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "NVIDIA GeForce RTX 4060 Laptop GPU", driver_name: "NVIDIA", driver_info: "537.53" }
2024-10-11T09:55:03.7130798+08:00 [INFO] Using surface present mode MAILBOX
2024-10-11T09:55:03.7173328+08:00 [WARN] Unable to forbid exclusive full screen
2024-10-11T09:55:03.7223389+08:00 [INFO] Creating texture 0x27941db4510 of size 1024x1024x1 and format R8Unorm, name 'atlas', handle 0
2024-10-11T09:55:03.7485809+08:00 [INFO] Creating buffer 0x2794239d0c0 of size 65536, name 'chunk-0', handle 1
2024-10-11T09:55:03.7588639+08:00 [INFO] Creating buffer 0x2794239b740 of size 4096, name 'chunk-0', handle 2
2024-10-11T09:55:03.7631721+08:00 [INFO] Creating buffer 0x2794239c940 of size 6832, name 'chunk-1', handle 3
2024-10-11T09:55:03.7701766+08:00 [INFO] set status on client 147755: Connecting
2024-10-11T09:55:03.8131221+08:00 [INFO] Using surface present mode MAILBOX
2024-10-11T09:55:03.8164949+08:00 [WARN] Unable to forbid exclusive full screen
2024-10-11T09:55:03.8212192+08:00 [INFO] Opening main db
2024-10-11T09:55:03.8254096+08:00 [INFO] Using Windows API for pseudoconsole
2024-10-11T09:55:03.8546573+08:00 [WARN] unhandled message: Metadata(SupermavenMetadataMessage { dust_strings: None })
2024-10-11T09:55:03.868934+08:00 [WARN] unhandled message: Set(SupermavenSetMessage { key: "disabled", value: Bool(false) })
2024-10-11T09:55:03.8691464+08:00 [INFO] Opening main db
2024-10-11T09:55:03.8839589+08:00 [INFO] Creating buffer 0x2794662aea0 of size 4096, name 'chunk-2', handle 4
2024-10-11T09:55:04.1711109+08:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-10-11T09:55:04.9010577+08:00 [INFO] add connection to peer
2024-10-11T09:55:04.9010988+08:00 [INFO] add_connection;
2024-10-11T09:55:04.9011507+08:00 [INFO] waiting for server hello
2024-10-11T09:55:04.9107165+08:00 [INFO] got server hello
2024-10-11T09:55:04.9107851+08:00 [INFO] set status to connected (connection id: ConnectionId { owner_id: 0, id: 0 }, peer id: PeerId { owner_id: 567, id: 679392 })
2024-10-11T09:55:04.9108452+08:00 [INFO] set status on client 147755: Connected { peer_id: PeerId { owner_id: 567, id: 679392 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-10-11T09:55:05.2200106+08:00 [WARN] unhandled message: TaskStatus(SupermavenTaskUpdateMessage { task: "Initializing", status: Complete, percent_complete: None })
2024-10-11T09:55:05.2200874+08:00 [WARN] unhandled message: ActiveRepo(SupermavenActiveRepoMessage { repo_simple_name: None })
2024-10-11T09:55:05.5297782+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.5301601+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.5301992+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.5302329+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.530348+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.5303799+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.530411+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.5304412+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.5304724+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.5450355+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:05.6829718+08:00 [INFO] Creating texture 0x27941db4040 of size 1024x1024x1 and format Bgra8UnormSrgb, name 'atlas', handle 5
2024-10-11T09:55:06.0575612+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.0723969+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.0729999+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.0730363+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.0730682+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.0731001+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.0731344+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.1356317+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.1356962+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.1357327+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.1358463+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.1358849+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:06.1359238+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:08.1504788+08:00 [INFO] Creating texture 0x27941db61f0 of size 1024x1024x1 and format R16Float, name 'atlas', handle 6
2024-10-11T09:55:10.2466756+08:00 [INFO] Creating buffer 0x27946628aa0 of size 6160, name 'chunk-3', handle 7
2024-10-11T09:55:10.2467321+08:00 [INFO] Creating buffer 0x2794662a2a0 of size 10528, name 'chunk-4', handle 8
2024-10-11T09:55:10.884589+08:00 [ERROR] Error { kind: Generic("Input watch path is neither a file nor a directory."), paths: [] }
2024-10-11T09:55:10.8868746+08:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "NVIDIA GeForce RTX 4060 Laptop GPU", driver_name: "NVIDIA", driver_info: "537.53" }
2024-10-11T09:55:10.8925151+08:00 [INFO] Using Windows API for pseudoconsole
2024-10-11T09:55:10.9203666+08:00 [ERROR] NotPresent
2024-10-11T09:55:10.9250306+08:00 [INFO] attempting to start language server "gopls", path: "E:\\code\\go\\learn", id: 1
2024-10-11T09:55:10.9276614+08:00 [INFO] found user-installed language server for gopls. path: "C:\\Users\\zero\\go\\bin\\gopls.exe", arguments: ["-mode=stdio"]
2024-10-11T09:55:10.9277774+08:00 [INFO] using project environment for language server LanguageServerName("gopls")
2024-10-11T09:55:10.9278585+08:00 [INFO] starting language server process. binary path: "C:\\Users\\zero\\go\\bin\\gopls.exe", working directory: "E:\\code\\go\\learn", args: ["-mode=stdio"]
2024-10-11T09:55:10.9934668+08:00 [WARN] unhandled message: Popup(SupermavenPopupMessage { message: "We don't have an active payment method on file for your account. To use Supermaven Pro, please add a payment method.", actions: [OpenUrl { label: "Add payment method", url: "https://supermaven.com/account" }, NoOp { label: "Dismiss" }] })
2024-10-11T09:55:11.0080034+08:00 [INFO] Creating buffer 0x27946629ca0 of size 7952, name 'chunk-5', handle 9
2024-10-11T09:55:11.0080862+08:00 [INFO] Creating buffer 0x27946627ba0 of size 12768, name 'chunk-6', handle 10
2024-10-11T09:55:11.2214389+08:00 [INFO] Creating buffer 0x2794618fd90 of size 13440, name 'chunk-7', handle 11
2024-10-11T09:55:11.4288167+08:00 [INFO] Creating buffer 0x27946190c90 of size 15120, name 'chunk-8', handle 12
2024-10-11T09:55:11.9566827+08:00 [WARN] unhandled capability registration: Registration { id: "workspace/didChangeConfiguration", method: "workspace/didChangeConfiguration", register_options: None }
2024-10-11T09:55:13.1877965+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:13.2191342+08:00 [WARN] request completed with error: failed to connect to the server
2024-10-11T09:55:15.2862279+08:00 [INFO] Creating buffer 0x27946191290 of size 15232, name 'chunk-9', handle 13
2024-10-11T09:55:26.1018363+08:00 [INFO] Creating buffer 0x27b59ccabc0 of size 12992, name 'chunk-10', handle 14
2024-10-11T09:55:26.9359257+08:00 [INFO] Creating buffer 0x279466299a0 of size 13888, name 'chunk-11', handle 15
2024-10-11T09:55:30.7278828+08:00 [INFO] Creating buffer 0x27b5646b370 of size 15456, name 'chunk-12', handle 16
2024-10-11T09:55:30.7451446+08:00 [INFO] Creating buffer 0x27b5646bf70 of size 15680, name 'chunk-13', handle 17
2024-10-11T09:55:30.75937+08:00 [INFO] Creating buffer 0x27b56470bf0 of size 16240, name 'chunk-14', handle 18
2024-10-11T09:55:30.7808286+08:00 [INFO] Creating buffer 0x27b5646dbf0 of size 16688, name 'chunk-15', handle 19
2024-10-11T09:55:30.7904641+08:00 [INFO] Creating buffer 0x27b5646fe70 of size 17024, name 'chunk-16', handle 20
2024-10-11T09:55:30.807956+08:00 [INFO] Creating buffer 0x27b5646b7f0 of size 17360, name 'chunk-17', handle 21
2024-10-11T09:55:30.8243689+08:00 [INFO] Creating buffer 0x27b564705f0 of size 18144, name 'chunk-18', handle 22
2024-10-11T09:55:30.8381944+08:00 [INFO] Creating buffer 0x27b5646b1f0 of size 18480, name 'chunk-19', handle 23
2024-10-11T09:55:30.855433+08:00 [INFO] Creating buffer 0x27b5646ccf0 of size 18928, name 'chunk-20', handle 24
2024-10-11T09:55:30.873161+08:00 [INFO] Creating buffer 0x27b564702f0 of size 19264, name 'chunk-21', handle 25
2024-10-11T09:55:30.8878327+08:00 [INFO] Creating buffer 0x27b56471970 of size 19376, name 'chunk-22', handle 26
2024-10-11T09:55:33.1194397+08:00 [INFO] Creating buffer 0x27b5646bdf0 of size 19712, name 'chunk-23', handle 27
2024-10-11T09:55:33.168878+08:00 [INFO] Creating buffer 0x27b5646c0f0 of size 20496, name 'chunk-24', handle 28
2024-10-11T09:55:33.1845639+08:00 [INFO] Creating buffer 0x27b5646def0 of size 20832, name 'chunk-25', handle 29
2024-10-11T09:55:46.2702639+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-10-11T09:55:46.2861935+08:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-10-11T09:55:46.3009698+08:00 [INFO] Creating a descriptor pool for at most 256 sets
2024-10-11T09:55:46.3166804+08:00 [INFO] Creating a descriptor pool for at most 256 sets
2024-10-11T09:55:46.6251465+08:00 [INFO] Creating buffer 0x27b5646fb70 of size 20608, name 'chunk-26', handle 30
2024-10-11T09:55:51.1366192+08:00 [INFO] Using surface present mode MAILBOX
2024-10-11T09:55:51.1387869+08:00 [WARN] Unable to forbid exclusive full screen
2024-10-11T09:55:51.1520017+08:00 [INFO] Creating buffer 0x27b56471370 of size 71792, name 'chunk-27', handle 31
2024-10-11T09:55:51.7522104+08:00 [INFO] Creating buffer 0x27b5646bc70 of size 78288, name 'chunk-28', handle 32
2024-10-11T09:55:51.7686734+08:00 [INFO] Creating buffer 0x27b5646d770 of size 79520, name 'chunk-29', handle 33
2024-10-11T09:55:51.7860187+08:00 [INFO] Creating buffer 0x27b5646c3f0 of size 83440, name 'chunk-30', handle 34
2024-10-11T09:55:51.8709278+08:00 [INFO] Creating buffer 0x27b5646e7f0 of size 89824, name 'chunk-31', handle 35
2024-10-11T09:55:53.6189763+08:00 [INFO] Creating buffer 0x27b5646d8f0 of size 89936, name 'chunk-32', handle 36
2024-10-11T09:55:53.6527461+08:00 [INFO] Creating buffer 0x27b5646da70 of size 90048, name 'chunk-33', handle 37
2024-10-11T09:55:53.7523678+08:00 [INFO] Creating buffer 0x27b5646eaf0 of size 90160, name 'chunk-34', handle 38
2024-10-11T09:55:54.1526518+08:00 [INFO] Creating buffer 0x27b5646ef70 of size 90272, name 'chunk-35', handle 39
2024-10-11T09:55:56.757282+08:00 [ERROR] failed to fetch cached embeddings via cloud model

Caused by:
    RPC request GetCachedEmbeddings failed: permission denied

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

--- a/crates/supermaven/src/supermaven.rs
+++ b/crates/supermaven/src/supermaven.rs
@@ -217,11 +217,11 @@ fn find_relevant_completion<'a>(
         };
 
         let current_cursor_offset = cursor_position.to_offset(buffer);
-        let original_cursor_offset = buffer.clip_offset(state.prefix_offset, text::Bias::Left);
-        if current_cursor_offset < original_cursor_offset {
+        if current_cursor_offset < state.prefix_offset {
             continue;
         }
 
+        let original_cursor_offset = buffer.clip_offset(state.prefix_offset, text::Bias::Left);
         let text_inserted_since_completion_request =
             buffer.text_for_range(original_cursor_offset..current_cursor_offset);
         let mut trimmed_completion = state_completion;

--- a/crates/supermaven/src/supermaven.rs
+++ b/crates/supermaven/src/supermaven.rs
@@ -217,7 +217,7 @@ fn find_relevant_completion<'a>(
         };
 
         let current_cursor_offset = cursor_position.to_offset(buffer);
-        let original_cursor_offset = state.prefix_offset;
+        let original_cursor_offset = buffer.clip_offset(state.prefix_offset, text::Bias::Left);
         if current_cursor_offset < original_cursor_offset {
             continue;
         }

